### ISSUE TITLE
Saving fails on some devices

### ISSUE THREAD

**Issue by mhtechdev** [MasayukiSuda/GPUVideo-android#41]
Saving a video using the MP4Composer fails on some devices, mostly Samsung Galaxy devices and Redmi. It ends in the OnFailed of the Mp4Composer, with exception.getMessage() returning null.

It also seems to only happen with some videos, on some devices. I have received a video which fails on one of those devices, but it works fine on my phone.

Any idea what could cause this? Unfortunately I don´t own any of those devices myself, so I am unable to dig deeper into the issue. It works fine on any device I have tested on. 
I am using the library in my app and some users have reported this problem.

**Comment by Hassaan68** [comment]
@mhtechdev hi. have you found any solution for this ? i am also facing this issue on samsung s6.

**Comment by mhtechdev** [comment]
No, unfortunately not. As I am unable to replicate it on any of my devices, it is hard for me to debug. Have you tried to figure out where exactly it fails?

**Comment by Hassaan68** [comment]
No, i am looking into it. the problem is that it fails randomly. not been able to find exact reason.

**Comment by mhtechdev** [comment]
Ok, please let me know if you find anything. Also, make sure that you dont use the same instance of filters you use for preview (if applicable), as that has been causing problems for me before.

**Comment by Hassaan68** [comment]
@mhtechdev thanks. i will inform you if i found anything helpful. actually i am using this library to watermark videos. new instance of GlWatermarkFilter and GlFilterGroup for each video
